.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,52 +1,7 @@
-<!--
+## Description
+<!-- Describe the changes. Link to any issues closed by this pull request. -->
 
-Thank you for contributing to cuDF :)
-
-Here are some guidelines to help the review process go smoothly.
-
-1. Please write a description in this text box of the changes that are being
-   made.
-
-2. Please ensure that you have written units tests for the changes made/features
-   added.
-
-3. There are CI checks in place to enforce that committed code follows our style
-   and syntax standards. Please see our contribution guide in `CONTRIBUTING.MD`
-   in the project root for more information about the checks we perform and how
-   you can run them locally.
-
-4. If you are closing an issue please use one of the automatic closing words as
-   noted here: https://help.github.com/articles/closing-issues-using-keywords/
-
-5. If your pull request is not ready for review but you want to make use of the
-   continuous integration testing facilities please mark your pull request as Draft.
-   https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft
-
-6. If your pull request is ready to be reviewed without requiring additional
-   work on top of it, then remove it from "Draft" and make it "Ready for Review".
-   https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request#marking-a-pull-request-as-ready-for-review
-
-   If assistance is required to complete the functionality, for example when the
-   C/C++ code of a feature is complete but Python bindings are still required,
-   then add the label `help wanted` so that others can triage and assist.
-   The additional changes then can be implemented on top of the same PR.
-   If the assistance is done by members of the rapidsAI team, then no
-   additional actions are required by the creator of the original PR for this,
-   otherwise the original author of the PR needs to give permission to the
-   person(s) assisting to commit to their personal fork of the project. If that
-   doesn't happen then a new PR based on the code of the original PR can be
-   opened by the person assisting, which then will be the PR that will be
-   merged.
-
-7. Once all work has been done and review has taken place please do not add
-   features or make changes out of the scope of those requested by the reviewer
-   (doing this just add delays as already reviewed code ends up having to be
-   re-reviewed/it is hard to tell what is new etc!). Further, please do not
-   rebase your branch on the target branch, force push, or rewrite history.
-   Doing any of these causes the context of any comments made by reviewers to be lost.
-   If conflicts occur against the target branch they should be resolved by
-   merging the target branch into the branch used for making the pull request.
```

Comment:
This part of the advice arguably could be ported into `CONTRIBUTING.md`. This seems like useful workflow information for contributors (especially new ones). IIUC, cudf uses squash-merges for PRs, so fixing up commits in response to review is not necessary for a clean commit history (and, as noted here, can hinder in the case of multiple rounds of reivew).

Reply:
I think that if it's outdated then there's no real reason to include it. I agree that github seems to be better than a few years ago at handling this kind of code movement (I don't recall having comments resolved by rebase and github losing track for a while).

Reply:
As I think about this more, this suggests an opportunity to rework some of the CONTRIBUTING guide. The differentiation between "[Your First Issue](https://github.com/rapidsai/cudf/blob/ad126065109aaa72b6eb324ba5abd555b70bb4ae/CONTRIBUTING.md#your-first-issue)" and "[Seasoned Developers](https://github.com/rapidsai/cudf/blob/ad126065109aaa72b6eb324ba5abd555b70bb4ae/CONTRIBUTING.md#seasoned-developers)" is irrelevant and a little presumptive. It'd be clearer to write something like "Development workflow" and "Project planning and organization", and incorporate tips about draft PRs, etc. To keep this focused, I will create a separate PR for the CONTRIBUTING guide before resolving this thread.

Reply:
https://stackoverflow.com/questions/50132325/github-advises-force-pushing-can-corrupt-your-pull-request-why

Comments are not lost when you force push, but they can be disassociated with the code when the commit they refer to is lost. You can no longer see the code they are referring to when you click on the file link at the top of the comment.  

I believe it would also be possible to lose changes from other contributors or changes made via the github UI (e.g. via code review suggestions). 

This is why I recommended we have a rule against force pushing once a PR is under review.  I believe that rule/suggestion should continue to be documented.

As the above linked SO answer suggests, 
> When you really need to do a force push, you can always create a new pull request, which is much safer and keeps away annoying problems

Reply:
I wish these templates and docs were RAPIDS-wide rather than specific to cuDF.